plugins/plugin-machine/che-plugin-machine-ext-client/src/main/java/org/eclipse/che/ide/extension/machine/client/machine/MachineStatusHandler.java

```diff
@@ -0,0 +1,143 @@
+                         final AppContext appContext,
+                         final EntityFactory entityFactory,
+                         final WorkspaceServiceClient workspaceServiceClient,
+                         final NotificationManager notificationManager,
+                         final MachineLocalizationConstant locale,
+                         final LoaderPresenter loader) {
+        this.eventBus = eventBus;
+        this.appContext = appContext;
+        this.entityFactory = entityFactory;
+        this.workspaceServiceClient = workspaceServiceClient;
+        this.notificationManager = notificationManager;
+        this.locale = locale;
+        this.loader = loader;
+
+        eventBus.addHandler(MachineStatusChangedEvent.TYPE, this);
+    }
+
+    @Override
+    public void onMachineStatusChanged(final MachineStatusChangedEvent event) {
+        final String machineName = event.getMachineName();
+        final String machineId = event.getMachineId();
+        final String workspaceId = event.getWorkspaceId();
+
+        workspaceServiceClient.getWorkspace(workspaceId).then(new Operation<WorkspaceDto>() {
+            @Override
+            public void apply(WorkspaceDto workspace) throws OperationException {
+                WorkspaceRuntimeDto workspaceRuntime = workspace.getRuntime();
+                if (workspaceRuntime == null) {
+                    return;
+                }
+
+                appContext.setWorkspace(workspace);
+
+                switch (event.getEventType()) {
+                    case CREATING:
+                        handleMachineCreating(machineId, workspaceRuntime);
+                        break;
+                    case RUNNING:
+                        handleMachineRunning(machineId, workspaceRuntime);
+                        break;
+                    case DESTROYED:
+                        notificationManager.notify(locale.notificationMachineDestroyed(machineName), SUCCESS, EMERGE_MODE);
+                        break;
+                    case ERROR:
+                        notificationManager.notify(event.getErrorMessage(), FAIL, EMERGE_MODE);
+                        break;
+                }
+            }
+        });
+    }
+
+    private MachineEntity getMachine(String machineId, WorkspaceRuntimeDto workspaceRuntime) {
+        for (MachineDto machineDto : workspaceRuntime.getMachines()) {
+            if (machineId.equals(machineDto.getId())) {
+                return entityFactory.createMachine(machineDto);
+            }
+        }
+        notificationManager.notify(locale.failedToFindMachine(machineId));
+        return null;
+    }
+
+    private void handleMachineCreating(final String machineId, final WorkspaceRuntimeDto workspaceRuntime) {
+        final MachineEntity machine = getMachine(machineId, workspaceRuntime);
+        if (machine == null) {
+            return;
+        }
+
+        if (machine.isDev()) {
+            // Will be used later
```

Comment:
@dkuleshov It came from here https://github.com/eclipse/che/pull/2461/files#diff-44495ab1a2865caef7c57f5fe6047c13L98
I'll remove it.